docs/searching-algorithms/interpolation-search.mdx

```diff
@@ -0,0 +1,276 @@
+<TabItem value="cpp">
+
+```cpp
+//INTERPOLATION SEARCH
+#include<bits/stdc++.h> //Importing library (This is the special type of library which will include all basic libraries)
+
+using namespace std;
+
+// If the searching element is present it will return its index, else it will return -1. 
+
+int interpolationSearch(int arr[], int n, int x) {
+  int low = 0, high = (n - 1); //Assigning lower index as 0 and higher index as n-1
+
+  while (low <= high && x >= arr[low] && x <= arr[high]) {
+    if (low == high) {
+      if (arr[low] == x) return low;
+      return -1;
+    }
+    // Probing the position with keeping uniform distribution in mind. 
+    int pos = low + (((double)(high - low) /
+      (arr[high] - arr[low])) * (x - arr[low]));
+
+    // Condition of element to be found 
+    if (arr[pos] == x)
+      return pos;
+
+    // If x is larger, x is in upper part 
+    if (arr[pos] < x)
+      low = pos + 1;
+
+    // If x is smaller, x is in the lower part 
+    else
+      high = pos - 1;
+  }
+  return -1;
+}
+
+int main() //Starting our main function 
+{
+
+  int num, x, loc;
+  std::cout << "Enter the size of array: ";
+  std::cin >> num;
+  int arr[num]; //Creating array of the size num
+  std::cout << "Enter elements: " << std::endl;
+
+  for (int i = 0; i < num; i++) //Enter the elements in the array
+  {
+    std::cin >> arr[i];
+  }
+
+  //Sorting our array
+
+  int n = sizeof(arr) / sizeof(arr[0]);
+
+
+  cout << "The elements entered are:" << std::endl;
+  cout << "[";
+  for (int i = 0; i < n; ++i) {
+    cout << arr[i];
+    if(i < n-1)
+      cout << ", ";
+  }
+  cout << "]";
+
+
+  sort(arr, arr + n);
+
+  cout << "\nThe sorted array is: " << std::endl;
+  cout << "[";
+  for (int i = 0; i < n; ++i) {
+    cout << arr[i];
+    if(i < n-1)
+      cout << ", ";
+  }
+  cout << "]";
+
+  //Entering the searching element
+  std::cout << "\n\nEnter the element to be searched: " << std::endl;
+  cin >> x; //Element to be searched 
+  int index = interpolationSearch(arr, n, x);
+  //If element was found 
+  if (index != -1)
+    cout << "Element found at index " << index; //Returning the index pf the element
+  else
+    cout << "Element not found."; //Returning -1 as element has not found 
+
+  return 0;
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Sample I/O  
+
+### Sample Input
+
+```
+5
+1
+5
+7
+2
+9
+7
+```
+
+### Sample Output
+
+```
+Enter the size of array : 5
+Enter elements :
+The elements entered are:
+[1, 5, 7, 2, 9]
+The sorted array is: 
+[1, 2, 5, 7, 9] 
+
+Enter the element to be searched: 7
+Element found at index 3
+```
+
+## Complexity Analysis
+
+n : number of the elements in the array
+
+### Time Complexity:
+
+- **Best Case**: O (1)
+- **Average Case**: O (log log n)
+- **Worst Case**: O (N)
+
+### Space Complexity: O (1)
+
+## Credits 
+- [Sharvari Raut](https://github.com/sharur7) for the [Python Implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/search/Interpolation_Search.py)
```

Comment:
`for the [Python Implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/search/Interpolation_Search.py) and the [C++ Implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/search/Interpolation_Search.cpp)`